.github/workflows/scanner-versioned-definitions-update.yaml

```diff
@@ -0,0 +1,224 @@
+    steps:
+    - name: Free up disk space
+      shell: bash
+      run: |
+        set +e
+        set -x
+        df -h
+        for delete in /usr/share/dotnet /usr/local/lib/android /opt/ghc /opt/hostedtoolcache/CodeQL; do
+          rm -rf "/mnt${delete:?}"
+        done
+        df -h
+
+    - name: Checkout specific reference
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        ref: ${{ env.ROX_PRODUCT_TAG }}
+
+    - uses: ./.github/actions/job-preamble
+      with:
+        gcp-account: ${{ secrets.GCP_SERVICE_ACCOUNT_STACKROX_CI }}
+
+    - name: Build updater
+      run: |
+        echo "Building updater for version ${{ env.ROX_BUNDLE_VERSION }} on branch ${{ env.ROX_PRODUCT_TAG }}..."
+        make tag
+        make -C scanner bin/updater
+
+    - name: Download NVD
+      uses: ./.github/actions/download-artifact-with-retry
+      with:
+        name: nvd
+        path: nvd.zip
+
+    - name: Set Updater
+      run: |
+        chmod +x scanner/bin/updater
+        mkdir vulns
+
+    - name: Create Folder
+      run: mkdir -p definitions/${{ env.ROX_BUNDLE_VERSION }}
+
+    - name: Run Updater (single bundle)
+      env:
+        STACKROX_NVD_ZIP_PATH: nvd.zip
+      run: |
+        scanner/bin/updater export vulns --manual-url "${{ needs.prepare-environment.outputs.manual_url }}"
+        mv vulns/* definitions/${{ env.ROX_BUNDLE_VERSION }}/
+
+    - name: Run updater (multi bundle)
+      env:
+        STACKROX_NVD_ZIP_PATH: nvd.zip
+      run: |
+        scanner/bin/updater export --manual-url "${{ needs.prepare-environment.outputs.manual_url }}" --split bundles
+        zip definitions/${{ env.ROX_BUNDLE_VERSION }}/vulnerabilities.zip bundles/*.json.zst
+
+    - name: Upload definitions artifacts
+      if: success() || failure()
+      uses: actions/upload-artifact@v4
+      with:
+        name: artifact_${{ env.ROX_BUNDLE_VERSION }}
+        path: definitions
+
+  upload-definitions:
+    needs:
+    - build-updater
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Download definitions artifacts
+      uses: actions/download-artifact@v4
+      with:
+        path: downloaded_artifacts
+
+    - name: Move files artifacts
+      run: rsync -av downloaded_artifacts/*/ definitions_files/
+
+    - name: Authenticate with Google Cloud
+      uses: google-github-actions/auth@v2
+      with:
+        credentials_json: ${{ secrets.GOOGLE_SA_STACKROX_HUB_VULN_DUMP_UPLOADER }}
+
+    - name: Set up Cloud SDK
+      uses: google-github-actions/setup-gcloud@v2
+
+    - name: Upload to GCS
+      run: |
+        ls -lr definitions_files
+        for dir in definitions_files/*; do
+          if [ -d "$dir" ]; then
+            gsutil -m cp -r "$dir" gs://definitions.stackrox.io/v4/vulnerability-bundles/
+          fi
+        done
+        gsutil cp -r gs://definitions.stackrox.io/v4/vulnerability-bundles/dev/* gs://definitions.stackrox.io/v4/vulnerability-bundles/1.0.0/
```

Comment:
Can we have a separate upload path for when this run from a pull request? See https://github.com/stackrox/stackrox/pull/12516 for an example.